### PR TITLE
Simplify device flow for notebooks

### PR DESCRIPTION
**Pull request by AhmetNSimsek**
```python
import siibra
siibra.fetch_ebrains_token(enable_flow=True) # or siibra.fetch_ebrains_token(True)
```

### CODE REVIEW

**Comment by xgui3783** [comment]
The fetch token method should be able to tell a user environment vs machine environment.

I hit a bump and wasn't able to distinguish notebook environment, but this could be useful : https://github.com/fastai/fastprogress/issues/26#issuecomment-464951467

File: siibra/retrieval/requests.py [hunk 1]
```diff
@@ -261,22 +261,29 @@ def init_oidc(cls):
...
         cls.device_flow()
 
+    @classmethod
+    def is_jupyter_notebook(cls):
+        """
+        Is the code running in the jupyter environment.
+        Includes -notebook, -lab, vscode notebook, and google colab notebooks. 
+        """
+        return True if any(k in ['JPY_INTERRUPT_EVENT', "JPY_PARENT_PID"] for k in os.environ) else False
+
```
Comment (xgui3783) [inline siibra/retrieval/requests.py:277]: probably doesn't need to be a whole function... see below

File: siibra/retrieval/requests.py [hunk 2]
```diff
@@ -261,22 +261,29 @@ def init_oidc(cls):
...
     @classmethod
     def set_token(cls, token):
         logger.info(f"Setting EBRAINS Knowledge Graph authentication token: {token}")
         cls._KG_API_TOKEN = token
 
     @classmethod
     def device_flow(cls):
+        if cls.is_jupyter_notebook():
+            os.environ["SIIBRA_ENABLE_DEVICE_FLOW"] = "1"
```
Comment (xgui3783) [inline siibra/retrieval/requests.py:286]: rather than check is notebook here... (see below)

**Comment by AhmetNSimsek** [comment]
> 

Thank you for the notes and the suggestion. I have committed a revision: now Siibra automatically checks if there is a Jupyter environment running including JupyterLab, VSCode Jupyter-Notebook, and Google Colab Notebooks.
The user only needs to use:
```python
siibra.fetch_ebrains_token()
```
as it would be in other environments.

File: siibra/retrieval/requests.py [hunk 3]
```diff
@@ -261,22 +261,29 @@ def init_oidc(cls):
...
 
         if not sys.__stdout__.isatty() and not os.getenv("SIIBRA_ENABLE_DEVICE_FLOW"):
```
Comment (xgui3783) [inline siibra/retrieval/requests.py:288]: add check is jupyter environment here...

in principle, this is where fail fast is checked (can be overwritten with env var)

**Review by xgui3783 — COMMENTED** [review]
some minor refactoring. otherwise, looks good

**Pull request merged**